lib/class-wp-rest-search-controller.php

```diff
@@ -0,0 +1,539 @@
+
+		if ( method_exists( $this, 'get_fields_for_response' ) ) {
+			$fields = $this->get_fields_for_response( $request );
+		} else {
+			$schema = $this->get_item_schema();
+			$fields = array_keys( $schema['properties'] );
+		}
+
+		// Base fields for every post.
+		$data = array();
+
+		if ( in_array( 'id', $fields, true ) ) {
+			$data['id'] = $post->ID;
+		}
+
+		if ( in_array( 'type', $fields, true ) ) {
+			$data['type'] = $post->post_type;
+		}
+
+		if ( in_array( 'link', $fields, true ) ) {
+			$data['link'] = get_permalink( $post->ID );
+		}
+
+		if ( in_array( 'title', $fields, true ) ) {
+			if ( post_type_supports( $post->post_type, 'title' ) ) {
+				add_filter( 'protected_title_format', array( $this, 'protected_title_format' ) );
+
+				$data['title'] = array(
+					'raw'      => $post->post_title,
+					'rendered' => get_the_title( $post->ID ),
+				);
+
+				remove_filter( 'protected_title_format', array( $this, 'protected_title_format' ) );
+			} else {
+				$data['title'] = array(
+					'raw'      => '',
+					'rendered' => '',
+				);
+			}
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		// Wrap the data in a response object.
+		$response = rest_ensure_response( $data );
+
+		$response->add_links( $this->prepare_links( $post ) );
+
+		return $response;
+	}
+
+	/**
+	 * Overwrites the default protected title format.
+	 *
+	 * By default, WordPress will show password protected posts with a title of
+	 * "Protected: %s". As the REST API communicates the protected status of a post
+	 * in a machine readable format, we remove the "Protected: " prefix.
+	 *
+	 * @since 2.9.0
+	 *
+	 * @return string Protected title format.
+	 */
+	public function protected_title_format() {
+		return '%s';
+	}
+
+	/**
+	 * Retrieves the item schema, conforming to JSON Schema.
+	 *
+	 * @since 2.9.0
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'search',
+			'type'       => 'object',
+			'properties' => array(
+				'id'    => array(
+					'description' => __( 'Unique identifier for the object.', 'gutenberg' ),
+					'type'        => 'integer',
+					'context'     => array( 'view', 'edit', 'embed' ),
+					'readonly'    => true,
+				),
+				'link'  => array(
+					'description' => __( 'URL to the object.', 'gutenberg' ),
+					'type'        => 'string',
+					'format'      => 'uri',
+					'context'     => array( 'view', 'edit', 'embed' ),
+					'readonly'    => true,
+				),
+				'title' => array(
+					'description' => __( 'The title for the object.', 'gutenberg' ),
+					'type'        => 'object',
+					'context'     => array( 'view', 'edit', 'embed' ),
+					'properties'  => array(
+						'raw'      => array(
```

Comment:
Is there a reason we need `context=edit` and `title.raw`? Or could we keep this controller as `context=embed,view` for now?